### PR TITLE
Remove related code about single file mode in DMFile

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #6233 #6426

Problem Summary:

1. Remove related code about single file mode in DMFile
2. remove sub_file_stats to cut down memory usage

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/SSTFilesToDTFilesOutputStream.cpp [hunk 1]
```diff
@@ -124,21 +124,8 @@ bool SSTFilesToDTFilesOutputStream<ChildStream>::newDTFileStream()
         return false;
     }
 
-    DMFileBlockOutputStream::Flags flags{};
-    switch (method)
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/SSTFilesToDTFilesOutputStream.cpp:128]: After removing these lines, the SSTFilesToDTFilesOutputStream::method is a useless private member and should be remove

**Review by JaySon-Huang — APPROVED** [review]
Rest LGTM

File: dbms/src/Server/DTTool/DTToolInspect.cpp [hunk 2]
```diff
@@ -81,43 +80,41 @@ int inspectServiceMain(DB::Context & context, const InspectArgs & args)
     if (args.check)
     {
         // for directory mode file, we can consume each file to check its integrity.
-        if (!dmfile->isSingleFileMode())
+
+        auto prefix = args.workdir + "/dmf_" + DB::toString(args.file_id);
```
Comment (Lloyd-Pottiger) [inline dbms/src/Server/DTTool/DTToolInspect.cpp:84]:
```suggestion
        auto prefix = fmt::format("{}/dmf_{}", args.workdir, args.file_id);
```

File: dbms/src/Server/DTTool/DTToolInspect.cpp [hunk 3]
```diff
@@ -81,43 +80,41 @@ int inspectServiceMain(DB::Context & context, const InspectArgs & args)
...
-            auto file = Poco::File{prefix};
-            std::vector<std::string> sub;
-            file.list(sub);
-            for (auto & i : sub)
+            if (endsWith(i, ".mrk") || endsWith(i, ".dat") || endsWith(i, ".idx") || i == "pack")
             {
-                if (endsWith(i, ".mrk") || endsWith(i, ".dat") || endsWith(i, ".idx") || i == "pack")
+                auto full_path = prefix;
+                full_path += "/";
+                full_path += i;
```
Comment (Lloyd-Pottiger) [inline dbms/src/Server/DTTool/DTToolInspect.cpp:94]: ditto

File: dbms/src/Server/DTTool/DTToolInspect.cpp [hunk 4]
```diff
@@ -81,43 +80,41 @@ int inspectServiceMain(DB::Context & context, const InspectArgs & args)
...
+                LOG_INFO(logger, "checking {}: ", i);
```
Comment (Lloyd-Pottiger) [inline dbms/src/Server/DTTool/DTToolInspect.cpp:95]: maybe also log dmfile filename?

File: dbms/src/Storages/DeltaMerge/File/DMFileReader.cpp [hunk 5]
```diff
@@ -51,67 +51,39 @@ DMFileReader::Stream::Stream(
...
     const ReadLimiterPtr & read_limiter)
-    : single_file_mode(reader.single_file_mode)
-    , avg_size_hint(reader.dmfile->getColumnStat(col_id).avg_size)
+    : avg_size_hint(reader.dmfile->getColumnStat(col_id).avg_size)
 {
     // load mark data
-    if (reader.single_file_mode)
-    {
-        auto mark_with_size_load = [&]() -> MarkWithSizesInCompressedFilePtr {
-            auto res = std::make_shared<MarkWithSizesInCompressedFile>(reader.dmfile->getPacks());
```
Comment (lidezhu) [inline dbms/src/Storages/DeltaMerge/File/DMFileReader.cpp:61]: `MarkWithSizesInCompressedFile` can also be removed.

**Review by lidezhu — APPROVED** [review]
LGTM with minor comment

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**